sync-online/service-worker.js

```diff
@@ -0,0 +1,147 @@
+// [Working example](/serviceworker-cookbook/sync-online/).
+
+var CACHE_NAME = 'dependencies-cache';
+
+// Files required to make this app work offline
+var REQUIRED_FILES = [
+  '/',
+  '/index.html',
+  'index.js'
+];
+
+// IndexedDB Settings
+var DATABASE_NAME = 'blog';
+var DATABASE_TABLE_NAME = 'posts';
+var DATABASE_POST_ID = '1';
+
+// Find indexedDB
+self.indexedDB = self.indexedDB || self.mozIndexedDB || self.webkitIndexedDB || self.msIndexedDB;
+if (self.indexedDB) {
+  // Initialize the database if necessary
+  // Calling this every time in case the user manually deleted DB
+  initializeDatabase();
+} else {
+  console.error('indexedDB not detected, this demo will not work as expected');
+}
+
+self.addEventListener('install', function(event) {
+  // Perform install step:  loading each required file into cache
+  event.waitUntil(
+    caches.open(CACHE_NAME)
+      .then(function(cache) {
+        // Add all offline dependencies to the cache
+        console.log('[install] Caches opened, adding all core components to cache');
+        return cache.addAll(REQUIRED_FILES);
+      })
+      .then(function() {
+        console.log('[install] All required resources have been cached, we\'re good!');
+        return self.skipWaiting();
+      })
+  );
+});
+
+self.addEventListener('fetch', function(event) {
+  // Only mock positive result if indexedDB is present
+  // and if they've POST'ed
+  if (self.indexedDB && event.request.url.indexOf('?') !== -1) {
+    // Save!
+    if (event.request.method === 'POST') {
+
+      // Send the fetch to the server if online
+      // We aren't concerned about the state of this because we have IndexDB as a backup
+      fetch(event.request.clone());
+
+      event.respondWith(
+        event.request.json().then(function(json) {
+          return managePostContent(json.content).then(function() {
+            return new Response(content, {
+              ok: true,
+              status: '200',
+              statusText: 'OK'
+            });
+          })
+        })
+      );
+    } else { // Get!
+      event.respondWith(
+        managePostContent(null).then(function(content) {
+          return new Response(JSON.stringify(content), {
+            ok: true,
+            status: '200',
+            statusText: 'OK'
+          });
+        })
+      );
+    }
+  } else {
+    // Make network request
+    event.respondWith(fetch(event.request));
+  }
+});
+
+self.addEventListener('activate', function(event) {
```

Comment:
I would tend to group `install` and `activate` code and keep them together with only one explanation for them.